### ISSUE TITLE
Tuple with '/r' as new line character

### ISSUE THREAD

**Issue by reconer** [IBMStreams/streamsx.tcp#28]
I am running the TCPServerDuplex (non-blob) example and I realize it cannot handle '\r' as new line. It only recognizes '\n'. 

As a result, the tuple data is not being fed downstream from TCPServer. It would be great if you can add support for such cases or better, a user defined end of message character. 

Thanks!

**Comment by leongor** [comment]
Can you give more details, please?
Do you send a message from external TCP client to TCPServer operator?
Does the messages end with '\r' without '\n' after it?

**Comment by reconer** [comment]
Yes thats right. its a tcp text stream whose end of line is a sentence with a '/r' at the end. it does not have the '/n' after it.

**Comment by chanskw** [comment]
@leongor Couple of questions:

1)  I think we hard-code "\n" as the line delimiter for a block of string.  Is this true?  
2)  Is it possible to add a parameter to specify a message delimiter for TCPServer operator?  For example, what if I don't want "\n" or "\r", but I want us to look at a different delimiter?
3)  In the case of blob, can we also have a customizable delimiter to break up the input?

Thanks!

**Comment by leongor** [comment]
@chanskw @reconer 
1) Yes, as a usual standard in TCP transmission.
2) Adding '\r' to be additional default line delimiter is quite easy. To have a parameter for a custom one (maybe even some regex) is more complicated, but also possible.
3) Not sure if it is wise to mix blocksize vs delimiter, but technically it's the same.

**Comment by chanskw** [comment]
@leongor can you show me where to look to override the default?  I can see if I can add a parameter.  Just want to try this and figure out what we need.

**Comment by leongor** [comment]
@chanskw here: [data_item.cpp](https://github.com/IBMStreams/streamsx.tcp/blob/master/com.ibm.streamsx.tcp/impl/src/mcts/data_item.cpp)
Look at function DataItem::addData()

**Comment by leongor** [comment]
@chanskw Did you have a chance to look at it?
